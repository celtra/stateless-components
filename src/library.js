import '@/stateless/define_helpers'

import * as filtersObject from '@/helpers/filters.js'
import * as directivesObject from '@/helpers/directives.js'
import * as itemsUtilsObject from '@/stateless/items_utils.js'

import CheckboxComponent from '@/stateless/checkbox.vue'
import DialogButtonComponent from '@/stateless/dialog_button.vue'
import DialogComponent from '@/stateless/dialog.vue'
import DropAreaComponent from '@/stateless/drop_area.vue'
import FileUploadRequirementsComponent from '@/stateless/file_upload_requirements.vue'
import FileUploadComponent from '@/stateless/file_upload.vue'
import ImageListComponent from '@/stateless/image_list.vue'
import InputComponent from '@/stateless/input.vue'
import MultiselectComponent from '@/stateless/multiselect.vue'
import RadioButtonComponent from '@/stateless/radiobutton.vue'
import SelectboxComponent from '@/stateless/selectbox.vue'
import SupportTextComponent from '@/stateless/support_text.vue'
import IconComponent from '@/stateless/icon.vue'
import ToastComponent from '@/stateless/Toast.vue'
import PieChartComponent from '@/stateless/PieChart.vue'
import InlineDialogComponent from '@/stateless/InlineDialog.vue'
import DefaultListComponent from '@/stateless/DefaultList.vue'
import DefaultListItemComponent from '@/stateless/DefaultListItem.vue'
import ScrollableListComponent from '@/stateless/ScrollableList.vue'
import TypeaheadComponent from '@/stateless/Typeahead.vue'
import TypeaheadMultiselectComponent from '@/stateless/TypeaheadMultiselect.vue'
import TooltipComponent from '@/stateless/Tooltip.vue'

export const filters = filtersObject
export const directives = directivesObject
export const itemsUtils = itemsUtilsObject

export const Checkbox = CheckboxComponent
export const DialogButton = DialogButtonComponent
export const Dialog = DialogComponent
export const DropArea = DropAreaComponent
export const FileUploadRequirements = FileUploadRequirementsComponent
export const FileUpload = FileUploadComponent
export const ImageList = ImageListComponent
export const Input = InputComponent
export const Multiselect = MultiselectComponent
export const RadioButton = RadioButtonComponent
export const Selectbox = SelectboxComponent
export const SupportText = SupportTextComponent
export const Icon = IconComponent
export const Toast = ToastComponent
export const PieChart = PieChartComponent
export const InlineDialog = InlineDialogComponent
export const DefaultList = DefaultListComponent
export const DefaultListItem = DefaultListItemComponent
export const ScrollableList = ScrollableListComponent
export const Typeahead = TypeaheadComponent
export const TypeaheadMultiselect = TypeaheadMultiselectComponent
export const Tooltip = TooltipComponent
