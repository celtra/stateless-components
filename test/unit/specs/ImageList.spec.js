import { ImageList } from '@/library.js'
import snapshotMixin from '../component_snapshot_mixin'

describe('ImageList', () => {
    snapshotMixin(ImageList)
})
