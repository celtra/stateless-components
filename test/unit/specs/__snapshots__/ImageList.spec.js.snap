// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList computed 126134677 animationRatio 1`] = `0`;

exports[`ImageList computed 126134677 buttons 1`] = `
Array [
  Object {
    "borderSpacing": 5,
    "isHovered": false,
    "size": 60,
  },
]
`;

exports[`ImageList computed 126134677 buttonsData 1`] = `
Array [
  Object {},
]
`;

exports[`ImageList computed 126134677 buttonsWidth 1`] = `61`;

exports[`ImageList computed 126134677 computedImages 1`] = `
Array [
  Object {
    "borderSpacing": 5,
    "hasWarning": undefined,
    "isCurrent": false,
    "isHovered": false,
    "loadingProgress": null,
    "offsetX": 0,
    "opacity": 0.8,
    "size": 60,
    "url": null,
  },
  Object {
    "borderSpacing": 5,
    "hasWarning": undefined,
    "isCurrent": false,
    "isHovered": false,
    "loadingProgress": null,
    "offsetX": 61,
    "opacity": 0.8,
    "size": 60,
    "url": null,
  },
  Object {
    "borderSpacing": 5,
    "hasWarning": undefined,
    "isCurrent": false,
    "isHovered": false,
    "loadingProgress": null,
    "offsetX": 122,
    "opacity": 0.8,
    "size": 60,
    "url": null,
  },
]
`;

exports[`ImageList computed 126134677 count 1`] = `2`;

exports[`ImageList computed 126134677 displayWidths 1`] = `
Array [
  60,
  60,
  60,
]
`;

exports[`ImageList computed 126134677 fullHeight 1`] = `67`;

exports[`ImageList computed 126134677 hasArrows 1`] = `
Object {
  "left": false,
  "right": false,
}
`;

exports[`ImageList computed 126134677 hoverIndex 1`] = `null`;

exports[`ImageList computed 126134677 imagesWidth 1`] = `121`;

exports[`ImageList computed 126134677 maxVisible 1`] = `7`;

exports[`ImageList computed 126134677 movingDirection 1`] = `null`;

exports[`ImageList computed 126134677 numVisible 1`] = `2`;

exports[`ImageList computed 126134677 targetWidth 1`] = `182`;

exports[`ImageList computed 126134677 targetWidths 1`] = `
Array [
  60,
  60,
  60,
]
`;

exports[`ImageList computed 126134677 targetXOffset 1`] = `0`;

exports[`ImageList computed 126134677 visibleWidths 1`] = `
Array [
  60,
  60,
  60,
]
`;
